### PR TITLE
Upgrade containerd dependency

### PR DESCRIPTION
**Pull request by lblackstone**

### CODE REVIEW

File: provider/go.mod [hunk 1]
```diff
@@ -31,6 +31,7 @@ require (
 )
 
 replace (
+	github.com/containerd/containerd => github.com/containerd/containerd v1.4.11 // Work around https://github.com/advisories/GHSA-c2h3-6mxw-7mvq
```
Comment (viveklak) [inline provider/go.mod:34]: Just curious if/when we can upgrade this 1.5.7+?

**Review by viveklak — APPROVED** [review]

File: provider/go.mod (hunk 1 above)
Comment (lblackstone) [reply provider/go.mod:34]: I'm not sure how to figure out which module is pulling in this dependency, so I just stuck with the same minor version. I noticed the same thing in pulumi/pulumi, so we may need to update a common dependency.

FYI @pulumi/platform

Comment (t0yv0) [reply provider/go.mod:34]:
```
anton@Antons-MacBook-Pro:~/pulumi-kubernetes/provider> go mod why github.com/containerd/containerd/sys 
go mod why github.com/containerd/containerd/sys 
# github.com/containerd/containerd/sys
github.com/pulumi/pulumi-kubernetes/provider/v3/pkg/provider
helm.sh/helm/v3/pkg/action
helm.sh/helm/v3/internal/experimental/registry
github.com/deislabs/oras/pkg/content
github.com/containerd/containerd/content/local
github.com/containerd/containerd/sys
```

Comment (lblackstone) [reply provider/go.mod:34]: Ah, my google-fu was letting me down yesterday. Thanks!

**Pull request merged**